File: apps/web/lib/customer-portal/wishlist.ts
import { portalApiFetch, withOrg } from '@/lib/customer-portal/api-client';

export interface WishlistItem {
  id: string;
  customer_id: string;
  product_id: string;
  variant_id?: string | null;
  notes?: string | null;
  created_at: string;
}

export interface WishlistListResponse {
  items: WishlistItem[];
  total: number;
  limit: number;
  offset: number;
}

export async function fetchMyWishlist(limit = 50, offset = 0): Promise<WishlistListResponse> {
  return portalApiFetch<WishlistListResponse>(withOrg('/api/v1/wishlist/', { limit: String(limit), offset: String(offset) }));
}

export async function removeMyWishlistItem(itemId: string): Promise<void> {
  await portalApiFetch(withOrg(`/api/v1/wishlist/${itemId}`), { method: 'DELETE' });
}

export async function moveMyWishlistItemToCart(itemId: string, sessionToken?: string, quantity = 1) {
  return portalApiFetch(withOrg(`/api/v1/wishlist/${itemId}/move-to-cart`, { session_token: sessionToken, quantity: String(quantity) }), { method: 'POST' });
}

File: apps/api/app/schemas/wishlist.py
from datetime import datetime

from pydantic import BaseModel, Field


class WishlistItemCreateRequest(BaseModel):
    product_id: str
    variant_id: str | None = None
    notes: str | None = None


class WishlistItemRead(BaseModel):
    id: str
    customer_id: str
    product_id: str
    variant_id: str | None = None
    notes: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class WishlistListResponse(BaseModel):
    items: list[WishlistItemRead]
    total: int
    limit: int
    offset: int
